lms/djangoapps/verify_student/views.py

```diff
@@ -856,6 +858,96 @@ def submit_photos_for_verification(request):
     return HttpResponse(200)
 
 
+def _compose_message_reverification_email(
+        course_key, user_id, relates_assessment, photo_verification, status, is_secure
+):  # pylint: disable=invalid-name
+    """ Composes subject and message for email
+
+    Args:
+        course_key(CourseKey): CourseKey object
+        user_id(str): User Id
+        relates_assessment(str): related assessment name
+        photo_verification(QuerySet/SoftwareSecure): A query set of SoftwareSecure objects or SoftwareSecure objec
+        status(str): approval status
+        is_secure(Bool): Is running on secure protocol or not
+
+    Returns:
+        None if any error occurred else Tuple of subject and message strings
+    """
+    try:
+        location_id = VerificationStatus.get_location_id(photo_verification)
+        usage_key = UsageKey.from_string(location_id)
+        course = modulestore().get_course(course_key)
+        redirect_url = get_redirect_url(course_key, usage_key.replace(course_key=course_key))
+
+        subject = render_to_string('emails/reverification_processed_subject.txt', {})
+        # Email subject must not contain newlines
+        subject = ''.join(subject.splitlines())
+
+        context = {
+            "status": status,
+            "course_name": course.display_name_with_default,
+            "assessment": relates_assessment,
+            "courseware_url": redirect_url
+        }
+
+        ver_block = modulestore().get_item(usage_key)
+        allowed_attempts = 1 if ver_block.attempts == 0 else ver_block.attempts
+        user_attempts = VerificationStatus.get_user_attempts(user_id, course_key, relates_assessment, location_id)
+        left_attempts = allowed_attempts - user_attempts
+        is_attempt_allowed = (allowed_attempts - user_attempts) > 0
+        current_date = timezone.now()
+        if ver_block.due:
+            verification_open = current_date <= ver_block.due
+        else:
+            verification_open = True
+
+        context["left_attempts"] = left_attempts
+        context["is_attempt_allowed"] = is_attempt_allowed
+        context["verification_open"] = verification_open
+        context["due_date"] = ver_block.due
+        context["is_secure"] = is_secure
+        context["site"] = microsite.get_value('SITE_NAME', 'localhost')
+
+        re_verification_link = reverse(
+            'verify_student_incourse_reverify',
+            args=(
+                unicode(course_key),
+                unicode(relates_assessment),
+                unicode(location_id)
+            )
+        )
+        context["reverify_link"] = re_verification_link
+        message = render_to_string('emails/reverification_processed.txt', context)
+        log_message = "Sending email to User_Id={}. Attempts left for this user are {}. " \
+                      "Allowed attempts {}. " \
+                      "Due Date {}".format(str(user_id), left_attempts, allowed_attempts, str(ver_block.due))
+        log.info(log_message)
```

Comment:
It's slightly more idiomatic to format this as:
```
log.info(
     "Sending email to User_Id=%s. Attempts left for this user are %s. "
     "Allowed attempts %s. "
     "Due Date %s",
     str(user_id), left_attempts, allowed_attempts, str(ver_block.due)
)
```
Log messages are the one case in which it's preferable to use %s instead of `.format()` for the very slight performance benefit of avoiding string concatenation if the log level is too low to actually be logged.